independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/multipart/MultipartPartReadingException.java

```diff
@@ -0,0 +1,19 @@
+package org.jboss.resteasy.reactive.server.multipart;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+/**
+ * Exception thrown when some part of the multipart input cannot be read using the appropriate
+ * {@link javax.ws.rs.ext.MessageBodyReader}.
+ * This exception is useful to application because it can be by an {@link javax.ws.rs.ext.ExceptionMapper} in order to customize
+ * the input error handling.
+ */
+public class MultipartPartReadingException extends WebApplicationException {
```

Comment:
Maybe extend from `javax.ws.rs.BadRequestException` so catch clauses and instanceof statements would work?